Library/Homebrew/cli/parser.rb

```diff
@@ -153,31 +156,54 @@ def summary
         @parser.to_s
       end
 
-      def parse(argv = @argv, allow_no_named_args: false)
+      def parse(argv = @argv, ignore_invalid_options: false)
         raise "Arguments were already parsed!" if @args_parsed
 
-        begin
-          named_args = @parser.parse(argv)
-        rescue OptionParser::InvalidOption => e
-          $stderr.puts generate_help_text
-          raise e
+        i = 0
+        remaining = []
+
+        argv, non_options = split_double_dash(argv)
+
+        while i < argv.count
```

Comment:
What's this all doing? It's not super obvious. Does it warrant a separate function and/or comment?

Reply:
Moved this into a separate method.

Reply:
As to what it does: Basically the same as `@parser.parse`, while being able to ignore invalid options.